core/src/test/java/org/apache/iceberg/TestCatalogUtil.java

```diff
@@ -193,26 +194,28 @@ public void buildCustomCatalog_withTypeSet() {
   public void loadCustomMetricsReporter_noArg() {
     Map<String, String> properties = Maps.newHashMap();
     properties.put("key", "val");
+    properties.put(
+        CatalogProperties.METRICS_REPORTER_IMPL, TestMetricsReporterDefault.class.getName());
 
-    MetricsReporter metricsReporter =
-        CatalogUtil.loadMetricsReporter(TestMetricsReporterDefault.class.getName());
+    MetricsReporter metricsReporter = CatalogUtil.loadMetricsReporter(properties);
     Assertions.assertThat(metricsReporter).isInstanceOf(TestMetricsReporterDefault.class);
   }
 
   @Test
   public void loadCustomMetricsReporter_badArg() {
-    Assertions.assertThatThrownBy(
-            () -> CatalogUtil.loadMetricsReporter(TestMetricsReporterBadArg.class.getName()))
-        .isInstanceOf(IllegalArgumentException.class)
-        .hasMessageContaining("missing no-arg constructor");
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(
+        CatalogProperties.METRICS_REPORTER_IMPL, TestMetricsReporterBadArg.class.getName());
+    Assertions.assertThat(CatalogUtil.loadMetricsReporter(properties))
+        .isInstanceOf(LoggingMetricsReporter.class);
   }
 
   @Test
   public void loadCustomMetricsReporter_badClass() {
-    Assertions.assertThatThrownBy(
-            () -> CatalogUtil.loadMetricsReporter(TestFileIONotImpl.class.getName()))
-        .isInstanceOf(IllegalArgumentException.class)
-        .hasMessageContaining("does not implement MetricsReporter");
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(CatalogProperties.METRICS_REPORTER_IMPL, TestFileIONotImpl.class.getName());
+    Assertions.assertThat(CatalogUtil.loadMetricsReporter(properties))
```

Comment:
nit: the properties could be inlined: `Assertions.assertThat(CatalogUtil.loadMetricsReporter(ImmutableMap.of(CatalogProperties.METRICS_REPORTER_IMPL, TestFileIONotImpl.class.getName())))`

Reply:
done